### ISSUE TITLE
Tree multiselect 'get_next_selected()' not collects all selected items

### ISSUE THREAD

**Issue by MikeSchulze** [godotengine/godot#42700]
**Godot version:**

Godot Engine v3.2.3.stable.mono.official - https://godotengine.org
OpenGL ES 3.0 Renderer: GeForce GTX 980/PCIe/SSE2

**OS/device including version:**
Windows 10 x64

**Issue description:**
The multi selection on Tree is broken when select multiple items from bottom to top in the tree.

**Steps to reproduce:**
To reproduce you have to select the items from bottom to top. (press CRTL + mouse )
* select folder 'folder_8'
* select folder 'folder_4'
* press right mouse button to show current selection
only folder_8 is found as selected

If you select from top to bottom it works as expected
* select folder 'folder_4'
* select folder 'folder_8'
* press right mouse button to show current selection
folder_4 and folder_8 is found as selected

**Suggested Feature**
Add a new function to get all selected items on SELECT_MULTI mode
`Array get_multi_selected() const`
Returns all selected items if SelectMode set to SELECT_MULTI otherwise null.
For SelectMode=SELECT_SINGLE use 'get_selected' instead

**Minimal reproduction project:**
[selectionBug.zip](https://github.com/godotengine/godot/files/5359758/selectionBug.zip)

**Comment by mrushyendra** [comment]
I believe the behavior is intentional, i.e. `get_next_selected` only traverses the tree from the top-down, so it is not able to get any selected nodes that come before the node that is supplied as the argument.

To get all the selected nodes, you can first pass `null` to `get_next_selected` to get the very first selected node:

```
	var selected_item:= get_next_selected(null)
	while selected_item:
		print(selected_item.get_text(0))
		selected_item = get_next_selected(selected_item)
```

**Comment by MikeSchulze** [comment]
@mrushyendra take you a look in the attached example project?
I use `var selected_item:= get_selected()` and not `get_next_selected` as first call

**Comment by mrushyendra** [comment]
Yeah, I was just suggesting a workaround if you want to get all the selected nodes. 

From the [documentation](https://docs.godotengine.org/en/stable/classes/class_tree.html#class-tree-method-get-next-selected) for `get_next_selected()`:
`If 'from' is null, this returns the first selected item.`

On the other hand, `get_selected()` may not return the first selected node. Instead it will return the node that is clicked first.

**Comment by MikeSchulze** [comment]
Thanks for your suggestion, i will change my code and use `get_next_selected(null)` to get first selected node ;)

**Comment by Kiisu-Master** [comment]
The workaround doesn't work.

My issue is slightly different: 
- when single selecting from bottom to top, the `get_next_selected()` will still report the bottom one as selected.

Using 4.2.1 stable.

**Comment by Kiisu-Master** [comment]
Update: 
I was able to fix the issue by "queueing" getting the selection for the next frame.
I noticed that the multiselection signal gets emitted multiple times and thought it may be the cause of the problem.
I don't know if there was a problem in my system with updating multiple times per frame, or the Tree needs a frame to update its selection stuff properly.

**Comment by MikeSchulze** [comment]
I'm closing it because it's been open for over 2 years.


In this case, I used `get_selected` where returns the selected focused item. So after two years looking into this issue, I say this is not a bug.

Instead, we should use `get_next_selected(null)` to find first selected item.
![Image](https://github.com/user-attachments/assets/dd153cda-8ac4-462d-a744-7171d249c462)

@Kiisu-Master please create a new bug issue for your threaded problem
